apps/birdbyt/birdbyt.star

```diff
@@ -0,0 +1,396 @@
+
+    response = http.get(url, params = params, headers = headers)
+
+    # e-bird API request failed
+    if response.status_code != 200:
+        return []
+
+    notable_sightings = response.json()
+    notable_list = [s.get("speciesCode") for s in notable_sightings]
+    print("number of notable sightings: ", len(notable_list))  # buildifier: disable=print
+
+    cache.set(cache_key, json.encode(notable_list), ttl_seconds = 1800)
+
+    return notable_list
+
+def get_recent_birds(params, ebird_key):
+    """Request a list of recent birds.
+
+    Args:
+      params: dictionary of parameters for the ebird API call
+      ebird_key: ebird API key
+
+    Returns:
+      ebird sightings data
+    """
+
+    # Do we already have cached data for this set of API params?
+    cache_key = "-".join(params.values())
+    sightings = cache.get(cache_key)
+    if sightings != None:
+        print("Cache hit:", cache_key)  # buildifier: disable=print
+        return json.decode(sightings)
+
+    # Nothing cached, so call the API
+    ebird_recent_obs_route = "/data/obs/geo/recent"
+    print("Cache miss:", cache_key, "\nCalling ", ebird_recent_obs_route)  # buildifier: disable=print
+    url = EBIRD_URL + ebird_recent_obs_route
+    headers = {"X-eBirdApiToken": ebird_key}
+
+    response = http.get(url, params = params, headers = headers)
+
+    # e-bird API request failed
+    if response.status_code != 200:
+        return [{
+            "comName": "Bird error!",
+            "locName": "API status code = " + str(response.status_code),
+        }]
+
+    sightings = response.json()
+    cache.set(cache_key, json.encode(sightings), ttl_seconds = 3600)
+    return sightings
+
+def parse_birds(sightings, tz):
+    """Parse ebird response data.
+
+    Args:
+      sightings: list of ebird sightings
+      tz: application's timezone
+
+    Returns:
+      a dictionary representing a single bird sighting
+    """
+
+    sighting = {}
+
+    number_of_sightings = len(sightings)
+    print("number of sightings: ", number_of_sightings)  # buildifier: disable=print
+
+    # request succeeded, but no birds found
+    if number_of_sightings == 0:
+        sighting = NO_BIRDS
+        return sighting
+
+    # grab a random bird sighting from ebird response
+    random_sighting = random.number(0, number_of_sightings - 1)
+    data = sightings[random_sighting]
+
+    sighting["bird"] = data.get("comName") or "Unknown bird"
+    sighting["loc"] = data.get("locName") or "Location unknown"
+    sighting["species"] = data.get("speciesCode") or "Unknown species code"
+    if data.get("obsDt"):
+        sighting["date"] = time.parse_time(data.get("obsDt"), format = "2006-01-02 15:04", location = tz)
+
+    return sighting
+
+def main(config):
+    """Update config.
+
+    Args:
+      config: config dict passed from the app
+
+    Returns:
+      rendered WebP image for Tidbyt display
+    """
+    ebird_key = secret.decrypt(EBIRD_API_KEY) or config.get("ebird_api_key")
```

Comment:
```suggestion
    random.seed(time.now().unix // 10)
    ebird_key = secret.decrypt(EBIRD_API_KEY) or config.get("ebird_api_key")
```

Reply:
This makes the app pick the same bird for all installations of the app over a 10 second window. Makes it easier for our backend to cache generated graphics properly. Hope you don't mind!

Reply:
@matslina quick question: will this make the app pick the same bird for all installations of the app even they're using different location parameters to get birds? or would it be the same bird for all installations that share the same config? Thanks!